Update SiteWrapper to be aware of LKSM App page.

labkey-danield opened the pull request:
#### Rationale
In order to run the LKSM automation against a trial instance the siteWrapper needs to be able to identify when a login has succeeded. The LKSM App page does not have the traditional LKS elements so added a check to for the user menu that is specific to the App UI. Hopefully it is general enough that if other apps are created that share the same header menu they will just work when running against a trial instance.

#### Related Pull Requests
* https://github.com/LabKey/sampleManagement/pull/564

#### Changes
* Update LabKeySiteWrapper.simpleSignIn to look for user menu in LKSM App.

labkey-danield committed:
Updated LabKeySiteWrapper.simpleSignIn to work with the LKSM app.
Added conditional check for a remote run before setting look and feel of the project.
Some opportunistic test logging/screenshots cleanup.

labkey-danield committed:
Merge branch 'develop' into fb_smTrialTestAutomation

labkey-tchad commented on src/org/labkey/test/LabKeySiteWrapper.java:
```diff
@@ -132,7 +132,8 @@ public void simpleSignIn()
             beginAt(WebTestHelper.buildURL("login", "login"));
-            waitForAnyElement("Should be on login or Home portal", Locator.id("email"), SiteNavBar.Locators.userMenu);
+            waitForAnyElement("Should be on login or Home portal", Locator.id("email"), SiteNavBar.Locators.userMenu,
+                    Locator.tagWithId("a", "user-menu-dropdown"));
```
Add a static `getLocator` method to `org.labkey.test.components.ui.navigation.UserMenu` rather than having this "user-menu-dropdown" id defined in several places.

labkey-danield replied:
Fixed, but not sure I fixed it the way you suggested. I would like to review this with you @labkey-tchad to make sure the fix is correct.

labkey-tchad commented on src/org/labkey/test/LabKeySiteWrapper.java:
```diff
@@ -154,6 +155,13 @@ public void simpleSignIn()
+                    goToHome();
+                    return true;
+                }
+
```
This should make sure the standard userMenu is present after navigating. Either remove the return line and move this `if` to the start of the `waitFor`; or have it assert that the standard userMenu is present after navigating home.

labkey-danield replied:
Fixed.

labkey-danield committed:
Slight modification to get a user id before impersonating a different user (impersonated user did not have permissions on a trial instance).
Move locator for the app user menu to a more central location.
Updated the helper code that will checkout a sample to take a user id as well as the name.

labkey-danield committed:
Implementing code review feedback.

labkey-danield committed:
Merge branch 'develop' into fb_smTrialTestAutomation

labkey-danield committed:
Merge branch 'develop' into fb_smTrialTestAutomation

labkey-tchad commented on src/org/labkey/test/components/ui/navigation/UserMenu.java:
```diff
@@ -45,4 +45,14 @@ protected Locator getToggleLocator()
+     *
+     * @return A locator to the drop down user menu button.
+     */
+    public static Locator AppUserMenu()
```
Java method names shouldn't be capitalized.

labkey-danield replied:
Yep, linter said the same thing. Fixed.

labkey-tchad commented on src/org/labkey/test/components/ui/navigation/UserMenu.java:
```diff
@@ -45,4 +45,14 @@ protected Locator getToggleLocator()
+     */
+    public static Locator AppUserMenu()
+    {
+        return Locator.tagWithId("a", "user-menu-dropdown");
```
Use the `MENU_ID` defined in this class.
```suggestion
        return Locator.tagWithId("a", MENU_ID);
```

Also, have `getToggleLocator` call and return this method.

labkey-danield replied:
Updated.

labkey-tchad reviewed (approved):
A couple of minor changes.

labkey-danield committed:
Implemented code review feedback.
